internal/services/cdn/cdn_endpoint_custom_domain_resource.go

```diff
@@ -159,3 +351,178 @@ func resourceArmCdnEndpointCustomDomainDelete(d *pluginsdk.ResourceData, meta in
+		return nil, err
+	}
+
+	output := &cdn.UserManagedHTTPSParameters{
+		CertificateSourceParameters: &cdn.KeyVaultCertificateSourceParameters{
+			OdataType:         utils.String("#Microsoft.Azure.Cdn.Models.KeyVaultCertificateSourceParameters"),
+			SubscriptionID:    &keyVaultId.SubscriptionId,
+			ResourceGroupName: &keyVaultId.ResourceGroup,
+			VaultName:         &keyVaultId.Name,
+			SecretName:        utils.String(raw["secret_name"].(string)),
+			SecretVersion:     utils.String(raw["secret_version"].(string)),
+			UpdateRule:        utils.String("NoAction"),
+			DeleteRule:        utils.String("NoAction"),
+		},
+		CertificateSource: cdn.CertificateSourceAzureKeyVault,
+		ProtocolType:      cdn.ProtocolTypeServerNameIndication,
+		MinimumTLSVersion: cdn.MinimumTLSVersionNone,
+	}
+
+	return output, nil
+}
+
+func flattenArmCdnEndpointCustomDomainCdnManagedHttpsSettings(input cdn.ManagedHTTPSParameters) []interface{} {
+	certificateType := ""
+	if params := input.CertificateSourceParameters; params != nil {
+		certificateType = string(params.CertificateType)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"certificate_type": certificateType,
+			"protocol_type":    string(input.ProtocolType),
+		},
+	}
+}
+
+func flattenArmCdnEndpointCustomDomainUserManagedHttpsSettings(input cdn.UserManagedHTTPSParameters) []interface{} {
+	var (
+		subscriptionId    string
+		resourceGroupName string
+		vaultName         string
+		secretName        string
+		secretVersion     string
+	)
+	if params := input.CertificateSourceParameters; params != nil {
+		if params.SubscriptionID != nil {
+			subscriptionId = *params.SubscriptionID
+		}
+		if params.ResourceGroupName != nil {
+			resourceGroupName = *params.ResourceGroupName
+		}
+		if params.VaultName != nil {
+			vaultName = *params.VaultName
+		}
+		if params.SecretName != nil {
+			secretName = *params.SecretName
+		}
+		if params.SecretVersion != nil {
+			secretVersion = *params.SecretVersion
+		}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"key_vault_id":   keyvaultParse.NewVaultID(subscriptionId, resourceGroupName, vaultName).ID(),
+			"secret_name":    secretName,
+			"secret_version": secretVersion,
+		},
+	}
+}
+
+func enableArmCdnEndpointCustomDomainHttps(ctx context.Context, client *cdn.CustomDomainsClient, id parse.CustomDomainId, params cdn.BasicCustomDomainHTTPSParameters) error {
+	_, err := client.EnableCustomHTTPS(ctx, id.ResourceGroup, id.ProfileName, id.EndpointName, id.Name, &params)
+	if err != nil {
+		return fmt.Errorf("sending enable request: %+v", err)
+	}
+
+	log.Printf("[DEBUG] Waiting for HTTPS to enable on %q", id)
+	deadline, _ := ctx.Deadline()
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{string(cdn.CustomHTTPSProvisioningStateEnabling)},
+		Target:     []string{string(cdn.CustomHTTPSProvisioningStateEnabled), string(cdn.CustomHTTPSProvisioningStateFailed), string(cdn.CustomHTTPSProvisioningStateDisabled)},
```

Comment:
if we're Enabling this then Failed and Disabled shouldn't be valid states?

```suggestion
		Target:     []string{string(cdn.CustomHTTPSProvisioningStateEnabled)},
```